packages/components/config/playwright/setup/utils/visual-regression.ts

```diff
@@ -47,6 +48,90 @@ class VisualRegression {
       });
     }
   }
+
+  /**
+   * Creates a sticker sheet on the page, grouping variants of components for visual regression testing.
+   *
+   * @param componentsPage - The page object used to interact with the components.
+   * @param componentTag - The tag name of the component to generate.
+   * @param attributes - Attributes to apply to the components, with key-value pairs representing attribute
+   *                     names and their possible values. The values are defined as an object of key-value pairs.
+   *
+   * @returns Locator for the component list containing all generated components.
+   */
+  async createStickerSheet(
+    componentsPage: ComponentsPage,
+    componentTag: string,
+    attributes: Record<string, Record<string, string>>,
+    children = '',
+  ) {
+    const generateComponentMarkup = () => {
+      const attributeEntries = Object.entries(attributes);
+      let markup = '';
+
+      if (attributeEntries.length === 0) {
+        return markup;
+      }
+
+      const [primaryAttribute, ...otherAttributes] = attributeEntries;
+      const [primaryKey, primaryValues] = primaryAttribute;
+
+      // eslint-disable-next-line
+      for (const primaryValue of Object.values(primaryValues)) {
+        let rowMarkup = '';
+
+        if (otherAttributes.length === 0) {
+          rowMarkup = `<div class="componentRowWrapper">
+            <${componentTag} ${primaryKey}="${primaryValue}">${children}</${componentTag}>
+          </div>`;
+        } else {
+          const combinations = otherAttributes.reduce(
+            (acc, [key, values]) => {
+              const valueArray = Object.values(values);
+              if (acc.length === 0) {
+                return valueArray.map((value) => `${key}="${value}"`);
+              }
+              const combined: string[] = [];
+              acc.forEach((prev) => {
+                valueArray.forEach((currVal) => {
+                  combined.push(`${prev} ${key}="${currVal}"`);
+                });
+              });
+              return combined;
+            },
+            [''],
+          );
+
+          rowMarkup = `<div class="componentRowWrapper">
+            ${combinations
+    .map(
+      (combination) =>
+        `<${componentTag} ${primaryKey}="${primaryValue}" ${combination}>${children}</${componentTag}>`,
+    )
+    .join('')}
+          </div>`;
+        }
+
+        markup += rowMarkup;
+      }
```

Comment:
```suggestion
 for (const primaryValue of Object.values(primaryValues)) {
        const combinations = otherAttributes.reduce<string[]>(
          (acc, [key, values]) => {
            const valueArray = Object.values(values);
            return acc.flatMap((prev) =>
              valueArray.map((currVal) => `${prev} ${key}="${currVal}"`));
          },
          [''],
        );

        const rowMarkup = `<div class="componentRowWrapper">
          ${combinations.map((combination) => `<${componentTag} 
        ${primaryKey}="${primaryValue}"${combination}>${children}</${componentTag}>`).join('')}
        </div>`;

        markup += rowMarkup;
      }
```

suggestion: i think we could simply this for-loop as suggested.